src/installer/tests/Microsoft.NET.HostModel.Tests/AppHost.Bundle.Tests/HammerServiceTest.cs

```diff
@@ -0,0 +1,100 @@
+                // On Unix systems, the servicing location is obtained from the environment variable $CORE_SERVICING.
+
+                return;
+            }
+
+            var fixture = sharedTestState.TestFixture.Copy();
+            var servicer = sharedTestState.ServiceFixture.Copy();
+
+            // Annotate the app as servicible, and then publish to a single-file.
+            string depsjson = BundleHelper.GetDepsJsonPath(fixture);
+            File.WriteAllText(depsjson, File.ReadAllText(depsjson).Replace("\"serviceable\": false", "\"serviceable\": true"));
+            var singleFile = BundleHelper.BundleApp(fixture);
+
+            // Create the servicing directory, and copy the servived DLL from service fixture to the servicing directory.
+            var serviceBasePath = Path.Combine(fixture.TestProject.ProjectDirectory, "coreservicing");
+            var servicePath = Path.Combine(serviceBasePath, "pkgs", BundleHelper.GetAppBaseName(servicer), "1.0.0");
+            Directory.CreateDirectory(servicePath);
+            File.Copy(BundleHelper.GetAppPath(servicer), Path.Combine(servicePath, BundleHelper.GetAppName(servicer)));
+
+            // Verify that the test DLL is loaded from the bundle when not being serviced
+            Command.Create(singleFile)
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdOutContaining("Hi Bellevue!");
+
+            // Verify that the test DLL is loaded from the servicing location when being serviced
+            Command.Create(singleFile)
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .EnvironmentVariable(BundleHelper.CoreServicingEnvVariable, serviceBasePath)
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdOutContaining("Hi Bengaluru!");
+        }
+
+        public class SharedTestState : IDisposable
+        {
+            public TestProjectFixture TestFixture { get; set; }
+            public TestProjectFixture ServiceFixture { get; set; }
+            public RepoDirectoriesProvider RepoDirectories { get; set; }
+
+            public SharedTestState()
+            {
+                RepoDirectories = new RepoDirectoriesProvider();
+                TestFixture = new TestProjectFixture("HammerServiceApp", RepoDirectories);
+                TestFixture
+                    .EnsureRestoredForRid(TestFixture.CurrentRid, RepoDirectories.CorehostPackages)
+                    .PublishProject(runtime: TestFixture.CurrentRid, 
+                                    outputDirectory: BundleHelper.GetPublishPath(TestFixture));
+
+                ServiceFixture = new TestProjectFixture("HammerServiceHelper", RepoDirectories, assemblyName: "Location");
```

Comment:
"HammerServicingHelper" -> "ServicedLocation", or "Location_Serviced" - to make it clear what it's for.